tools/crm_node.c

```diff
@@ -526,6 +526,39 @@ print_quorum(void)
     exit_code = pcmk_rc2exitc(rc);
 }
 
+/*!
+ * \internal
+ * \brief Extend a transaction by removing a node from a CIB section
+ *
+ * \param[in,out] cib        Active CIB connection
+ * \param[in]     element    CIB element containing node name and/or ID
+ * \param[in]     section    CIB section that \p element is in
+ * \param[in]     node_name  Name of node to purge (NULL to leave unspecified)
+ * \param[in]     node_id    Node ID of node to purge (0 to leave unspecified)
+ *
+ * \note At least one of node_name and node_id must be specified.
+ * \return Standard Pacemaker return code
+ */
+static int
+remove_from_section(cib_t *cib, const char *element, const char *section,
+                    const char *node_name, long node_id)
+{
+    xmlNode *xml = NULL;
+    int rc = pcmk_rc_ok;
+
+    xml = create_xml_node(NULL, element);
+    if (xml == NULL) {
+        return pcmk_rc_error;
+    }
+    crm_xml_add(xml, XML_ATTR_UNAME, node_name);
+    if (node_id > 0) {
+        crm_xml_set_id(xml, "%ld", node_id);
```

Comment:
`crm_xml_add()` would be fine since we definitely don't need to sanitize an integer; also fine to keep this for consistency if you want

Reply:
we need the %ld

Reply:
oh yeah